src/pip/_internal/commands/check.py

```diff
@@ -26,18 +26,16 @@ def run(self, options, args):
         for project_name in missing:
             version = package_set[project_name].version
             for dependency in missing[project_name]:
-                write_output(
-                    "%s %s requires %s, which is not installed.",
-                    project_name, version, dependency[0],
-                )
+                msg = "{} {} requires {}, which is not installed.".format(
+                    project_name, version, dependency[0])
+                write_output(msg)
```

Comment:
`write_output()` uses `logger.info()`, don’t replace arguments passed to it. Same for all the later `write_output()` calls.